scala3doc/src/dotty/dokka/tasty/TastyParser.scala

```diff
@@ -25,79 +25,7 @@ import scala.util.Try
   *
   * Delegates most of the work to [[TastyParser]] [[dotty.dokka.tasty.TastyParser]].
   */
-case class DokkaTastyInspector(sourceSet: SourceSetWrapper, parser: Parser, config: DottyDokkaConfig) extends DokkaBaseTastyInspector with TastyInspector
-
-import dotty.tools.dotc.core.Contexts.{Context => DottyContext}
-case class SbtDokkaTastyInspector(
-  sourceSet: SourceSetWrapper,
-  config: DottyDokkaConfig,
-  filesToDocument: List[String],
-  rootCtx: DottyContext,
-) extends DokkaBaseTastyInspector:
-  self =>
-
-  import dotty.tools.dotc.Compiler
-  import dotty.tools.dotc.Driver
-  import dotty.tools.dotc.Run
-  import dotty.tools.dotc.core.Contexts.Context
-  import dotty.tools.dotc.core.Mode
-  import dotty.tools.dotc.core.Phases.Phase
-  import dotty.tools.dotc.fromtasty._
-  import scala.quoted.runtime.impl.QuotesImpl
-
-
-  val parser: Parser = null
-
-  def run(): List[DPackage] = {
-    val driver = new InspectorDriver
-    driver.run(filesToDocument)(rootCtx)
-    result()
-  }
-
-  class InspectorDriver extends Driver:
-    override protected def newCompiler(implicit ctx: Context): Compiler = new TastyFromClass
-
-    def run(filesToDocument: List[String])(implicit ctx: Context): Unit =
-      doCompile(newCompiler, filesToDocument)
-
-  end InspectorDriver
-
-  class TastyFromClass extends TASTYCompiler:
-
-    override protected def frontendPhases: List[List[Phase]] =
-      List(new ReadTasty) :: // Load classes from tasty
-      Nil
-
-    override protected def picklerPhases: List[List[Phase]] = Nil
-
-    override protected def transformPhases: List[List[Phase]] = Nil
-
-    override protected def backendPhases: List[List[Phase]] =
-      List(new TastyInspectorPhase) ::  // Print all loaded classes
-      Nil
-
-    override def newRun(implicit ctx: Context): Run =
-      reset()
-      new TASTYRun(this, ctx.fresh.addMode(Mode.ReadPositions).addMode(Mode.ReadComments))
-
-  end TastyFromClass
-
-  class TastyInspectorPhase extends Phase:
-
-    override def phaseName: String = "tastyInspector"
-
-    override def run(implicit ctx: Context): Unit =
-      val qctx = QuotesImpl()
-      self.processCompilationUnit(using qctx)(ctx.compilationUnit.tpdTree.asInstanceOf[qctx.reflect.Tree])
-
-  end TastyInspectorPhase
-
-end SbtDokkaTastyInspector
-
-trait DokkaBaseTastyInspector:
-  val sourceSet: SourceSetWrapper
-  val parser: Parser
-  val config: DottyDokkaConfig
+case class DokkaTastyInspector(sourceSet: SourceSetWrapper, parser: Parser, config: DottyDokkaConfig) extends TastyInspector:
```

Comment:
We shouldn't delete this. When we're called from SBT, we should run TastyInspector with the `Context` that SBT gives us, or otherwise our errors won't be forwarded to SBT.

Reply:
I've fixed the problem the other way around and we provide our own context to TastyInspector (I needed to change API a bit)